### PR TITLE
Support selecting files with globs

### PR DESCRIPTION
**Pull request by mkhl**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -70,6 +72,11 @@ function filterInvisibleFiles(files) {
 	return files.filter(file => !file.startsWith("."));
 }
 
+function selectFilesMatching(pattern) {
+	let patterns = Array.isArray(pattern) ? pattern : !pattern ? [] : [pattern];
```
Comment (moonglum) [inline lib/index.js]: For my taste there is too much happening in this one line. But it is right at my threshold, so I'm ok with it.

Comment (FND) [reply lib/index.js]: Nested ternaries are _very_ rarely acceptable (if ever) - and then only with extra parentheses.

This doesn't strike me as a justifiable case here.

Comment (mkhl) [reply lib/index.js]: How’d you like it like this:
```js
function asArray(things) {
  if(!thing) {
    return [];
  }
  if(Array.isArray(things)) {
    return things;
  }
  return [things];
}
```

Comment (FND) [reply lib/index.js]: I like the [early returns](http://rikschennink.nl/thoughts/the-bouncer-pattern/), but would keep the ternary for arrayification:

```javascript
function asArray(value) {
    if(!value) {
        return [];
    }

    return Array.isArray(value) ? value : [value];
}
```
(note the variable name and whitespace)

However, I'm not sure

* whether falsiness is what we want in that first line (I don't know that fully grok how this all fits together; I'm just here for the aesthetics)
* whether a separate function is warranted

Comment (mkhl) [reply lib/index.js]: > note the … whitespace
The blank line?  Um, OK.

> whether falsiness is what we want in that first line

The argument here is the `match` property of a static configuration object.
We need an array of patterns, so we have three cases to handle:

*	`match` is missing, copy all files. Example:
	```js
	{
		source: …,
		target: …
	}
	```

*	`match` is a pattern, copy files matching it.  Example:
	```js
	{
		source: …,
		target: …,
		match: "foo.bar/baz"
	}
	```

*	`match` is an array of patterns, copy files matching all.  Example:
	```js
	{
		source: …,
		target: …,
		match: ["*.foo", "*.bar", "!baz/*"]
	}
	```

> whether a separate function is warranted

I prefer separate functions over uninitialised variables assigned in some (and hopefully all) branches of some conditional.

**Pull request closed without merging**